apps/api/src/tasks/import-transactions.task.ts

```diff
@@ -0,0 +1,404 @@
+        endpoint + `/${ibanAccount.id}` + `?dateFrom=${today}&dateTo=${today}`,
+        {
+          headers: {
+            'x-user-hash': this.userHash,
+            'x-agent-hash': this.agentHash,
+          },
+        },
+      )
+    ).data
+
+    return response.transactions
+  }
+
+  // Checks to see if all transactions have been processed already
+  private async hasNewOrNonImportedTransactions(transactions: IrisTransactionInfo[]) {
+    const transactionIds = transactions.map((trx) => trx.transactionId)
+
+    const count = await this.prisma.bankTransaction.count({
+      where: {
+        id: {
+          in: transactionIds,
+        },
+      },
+    })
+
+    return transactionIds.length === count
+  }
+
+  // Only prepares the data, without inserting it in the DB
+  private prepareBankTransactionRecords(
+    transactions: IrisTransactionInfo[],
+    ibanAccount: IrisIbanAccountInfo,
+  ) {
+    const data: Prisma.BankTransactionCreateManyInput[] = transactions.map((trx) => ({
+      id: trx.transactionId?.trim() || ``,
+      ibanNumber: ibanAccount.iban,
+      bankName: ibanAccount.bankName,
+      bankIdCode: this.bankBIC,
+      transactionDate: new Date(trx.valueDate),
+      senderName: trx.debtorName,
+      recipientName: trx.creditorName,
+      senderIban: trx.debtorAccount?.iban,
+      recipientIban: trx.creditorAccount?.iban,
+      type: trx.creditDebitIndicator === 'CREDIT' ? 'credit' : 'debit',
+      amount: toMoney(trx.transactionAmount.amount),
+      currency: trx.transactionAmount.currency,
+      description: trx.remittanceInformationUnstructured.trim(),
+    }))
+
+    return data
+  }
+
+  private async processDonations(bankTransactions: Prisma.BankTransactionCreateManyInput[]) {
+    const processedBankTransactions: Prisma.BankTransactionCreateManyInput[] = []
+
+    // Try to recognize campaign payment references
+    const matchedPaymentRef: string[] = []
+    bankTransactions.forEach((trx) => {
+      if (trx.type !== 'credit' || trx.description === 'STRIPE') return
+
+      const matchedRef = trx.description.replace(/[ _]+/g, '-').match(this.regexPaymentRef)
+      if (matchedRef) matchedPaymentRef.push(matchedRef[0])
+    })
+
+    /* 
+     Better get all campaigns in a single query
+     than execute a separate one for each transaction -
+     more performent and reliable approach
+    */
+    const campaigns = await this.prisma.campaign.findMany({
+      where: {
+        paymentReference: {
+          in: matchedPaymentRef,
+        },
+      },
+      include: {
+        vaults: true,
+      },
+    })
+
+    for (const trx of bankTransactions) {
+      // We're interested in parsing only incoming trx's
+      if (trx.type !== 'credit') {
+        processedBankTransactions.push(trx)
+        continue
+      }
+
+      // Stripe payments should not be parsed
+      if (trx.description === 'STRIPE') {
+        processedBankTransactions.push(trx)
+        continue
+      }
```

Comment:
You are right! Here I'm using an additional array to store the transaction objects passed as parameters, but since objects are passed by reference and not by value, the additional array `processedBankTransactions` turns out to be just an array of pointers. So I removed that one to save on memory and I update the donation status directly in the passed transactions